overlord/snapstate/snapstate.go

```diff
@@ -2183,6 +2061,52 @@ func autoRefreshPhase2(ctx context.Context, st *state.State, updates []*refreshC
 	return tasksets, nil
 }
 
+// checkDiskSpace checks if there is enough space for the requested snaps and their prerequisites
+func checkDiskSpace(st *state.State, changeKind string, infos []minimalInstallInfo, userID int) error {
+	var featFlag features.SnapdFeature
+
+	switch changeKind {
+	case "install":
+		featFlag = features.CheckDiskSpaceInstall
+	case "refresh":
+		featFlag = features.CheckDiskSpaceRefresh
+	default:
+		panic(fmt.Sprintf("unsupported disk space check for change of kind %q", changeKind))
+	}
+
+	tr := config.NewTransaction(st)
+	enabled, err := features.Flag(tr, featFlag)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	} else if !enabled {
```

Comment:
(nitpick) Could we move the `if !enabled` out of the else into the next line? This would make reading the code easier as one does not have to think about it less (at least I find it easier to read). Also do we have a test where the option is set available but set to "enabled=false" ?

Reply:
> (nitpick) Could we move the if !enabled out of the else into the next line? This would make reading the code easier as one does not have to think about it less (at least I find it easier to read).

:+1: 

> Also do we have a test where the option is set available but set to "enabled=false" ?

Yes, I I think features_test.go tests the recognised values for these flags